Xamarin.Essentials/Connectivity/Connectivity.ios.cs

```diff
@@ -13,6 +14,22 @@ static void StartListeners()
             listener.ReachabilityChanged += OnConnectivityChanged;
         }
 
+        static SignalStrength PlatformSignalStrength()
+        {
+            var signalstrength = (int)new NEHotspotNetwork().SignalStrength * 3;
+            switch (signalstrength)
+            {
+                case 1:
+                    return SignalStrength.Weak;
+                case 2:
+                    return SignalStrength.Fair;
+                case 3:
+                    return SignalStrength.Strong;
+                default:
+                    throw new InvalidOperationException("Exhaustive switch statement hit default value.");
```

Comment:
Maybe rather than throwing, since this comes from the platform and we have no control, just return unknown?

Reply:
Here I disagree. If the platform spec changes our results become worthless and the exception helps us find it. If our logic changes the switch statement should be adapted to be exhaustive. This exception should never be hit ever on iOS

Reply:
Well the question would be whether we would rather return senseless values on an api change or throw an Exception. I believe this to be a stable api though, so this should be dead code´.